code/game/objects/items/policetape.dm

```diff
@@ -125,6 +125,10 @@ var/list/tape_roll_applications = list()
 	if(istype(A, /obj/machinery/door/airlock))
 		var/turf/T = get_turf(A)
 		var/obj/item/tape/P = new tape_type(T.x,T.y,T.z)
+		var/obj/machinery/door/airlock/R = A
+		if(R.is_emagged())
+			to_chat(user, "You cannot tape this door.")
+			return // we don't want problems with density
```

Comment:
The door state doesn't update when it's fixed while it's taped, that's the main problem. This just makes it so you cannot tape an emagged door.